### ISSUE TITLE
Issue with save_mtx function

### ISSUE THREAD

**Issue by mabbasi6** [QI2lab/merfish3d-analysis#34]
Hi @dpshepherd,

I ran decoding and noticed that in the datastore.save_mtx section I was getting an error raised for the create_mtx function from dataio. Here is the error:

```
Traceback (most recent call last):
  File "/home/momo/miniforge3/envs/wf-merfish/lib/python3.10/site-packages/pandas/core/indexes/base.py", line 3805, in get_loc
    return self._engine.get_loc(casted_key)
  File "index.pyx", line 167, in pandas._libs.index.IndexEngine.get_loc
  File "index.pyx", line 196, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 2606, in pandas._libs.hashtable.Int64HashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 2630, in pandas._libs.hashtable.Int64HashTable.get_item
KeyError: 1963

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/momo/Repos/FISH-analysis/03_MERFISH_pixeldecode_qi2lab.py", line 42, in <module>
    decode_pixels()
  File "/home/momo/Repos/FISH-analysis/03_MERFISH_pixeldecode_qi2lab.py", line 38, in decode_pixels
    datastore.save_mtx()
  File "/home/momo/Repos/wf-merfish/merfish3d-analysis/src/merfish3danalysis/qi2labDataStore_baysor.py", line 3083, in save_mtx
    create_mtx(
  File "/home/momo/Repos/wf-merfish/merfish3d-analysis/src/merfish3danalysis/utils/_dataio.py", line 208, in create_mtx
    matrix.at[feature_to_index[feature], cell] += 1
  File "/home/momo/miniforge3/envs/wf-merfish/lib/python3.10/site-packages/pandas/core/indexing.py", line 2575, in __getitem__
    return super().__getitem__(key)
  File "/home/momo/miniforge3/envs/wf-merfish/lib/python3.10/site-packages/pandas/core/indexing.py", line 2527, in __getitem__
    return self.obj._get_value(*key, takeable=self._takeable)
  File "/home/momo/miniforge3/envs/wf-merfish/lib/python3.10/site-packages/pandas/core/frame.py", line 4214, in _get_value
    series = self._get_item_cache(col)
  File "/home/momo/miniforge3/envs/wf-merfish/lib/python3.10/site-packages/pandas/core/frame.py", line 4638, in _get_item_cache
    loc = self.columns.get_loc(item)
  File "/home/momo/miniforge3/envs/wf-merfish/lib/python3.10/site-packages/pandas/core/indexes/base.py", line 3812, in get_loc
    raise KeyError(key) from err
KeyError: 1963
```

The error points to the first cell ID (1963). I think it is coming from line 188 in the _dataio.py where it uses ```cells = transcripts_df["cell"].dropna().unique()[1:]``` to discard the first entry, which assumes is 0; but the first entry is actually not 0, and it's 1963 in my data. Instead, using this drops the 0 value, independent of the location.
```
cells = transcripts_df["cell"].dropna().unique()
cells = cells[cells != 0]
```

Thank you

**Comment by dpshepherd** [comment]
Thanks. You should still have the decoded results and can look at them without the MTX. In particular, the diagnostic files in the segmentation directory are quite helpful. 

I haven't worked on mtx creation since the Baysor update, so you are likely correct in your suggestion that we need to update that code.

**Comment by dpshepherd** [comment]
Closed by #35
